src/services/account/cleeng.account.service.ts

```diff
@@ -0,0 +1,296 @@
+import jwtDecode from 'jwt-decode';
+import { inject, injectable } from 'inversify';
+
+import type AccountService from './account.service';
+
+import type CleengService from '#src/services/cleeng/cleeng.service';
+import type { Config } from '#types/Config';
+import { getOverrideIP } from '#src/utils/common';
+import type {
+  ChangePassword,
+  GetCustomer,
+  GetCustomerConsents,
+  GetPublisherConsents,
+  Login,
+  Register,
+  ResetPassword,
+  UpdateCustomer,
+  UpdateCustomerConsents,
+  GetCaptureStatus,
+  UpdateCaptureAnswers,
+  AuthData,
+  JwtDetails,
+  GetCustomerConsentsResponse,
+  GetCaptureStatusResponse,
+  Capture,
+  GetLocales,
+  LoginPayload,
+  RegisterPayload,
+  UpdateCaptureAnswersPayload,
+  UpdateCustomerConsentsPayload,
+  UpdateCustomerPayload,
+  ChangePasswordWithOldPassword,
+  UpdatePersonalShelves,
+  NotificationsData,
+  ExportAccountData,
+  SocialURLSData,
+  DeleteAccount,
+} from '#types/account';
+import { SERVICES } from '#src/ioc/types';
+
+@injectable()
+export default class CleengAccountService implements AccountService {
+  private cleengService: CleengService;
+
+  public canUpdateEmail = true;
```

Comment:
Since we're doing this refactor, it would be nice to use types to actually couple the implementation to the 'feature support' for each service. That would mean, instead of flags and throwing 'not implemented exceptions', we would make those methods be optional and if a service has the method, then it can be assumed that it supports the functionality.
```
if (accountService.deleteAccount) {
  accountService.deleteAccount();
}
```

Reply:
I think we still need some feature flags so the UI can respond accordingly. E.g. hide the delete account button when the service doesn't support account deletion.

But I do agree this can be done cleaner with, for example, a `hasFeature(feature: string)` method on the AccountService interface.

Reply:
The UI can inspect the service just as well.
```
<>
  {accountController.exportUserData && <ExportUserDataSection />}
</>
```

Reply:
Yes, but then you would need to make the method in the controller optional which depends on the actual service being used. The UI shouldn't care which service is used by the controller e.g. JWP/Cleeng.